### PR TITLE
Shrink the API, add resources

### PR DESCRIPTION
**Pull request by ielis**
Hi @belkassaby ,
I'd like to make these changes to shrink the API & add some `DownloadableResource`s. 

The changes include:
- URLs are not hardcoded into Java code but externalized into `resource-url.properties` file.
- There is no way to instantiate the `BioDownloaderBuilder` except from the `BioDownloaderBuilder.builder(...)`.
- `DownloadableResource` and is not part of the API.
- No duplicate resources can be downloaded, requesting duplicated resource throws an error.
- GENCODE GTF files are added into the downloadable resources.
- Nicer logging without dangling `\n` characters.

### CODE REVIEW

**Review by pnrobinson — APPROVED** [review]
makes sense

File: src/main/java/org/monarchinitiative/biodownload/ftp/NIOftp.java [hunk 1]
```diff
@@ -66,13 +66,11 @@ public static File ftp(final URL from, File target) throws FileDownloadException
...
     }
 
-    private static FileReader createReader(final URL from) throws FileDownloadException {
-        assert !Objects.isNull(from);
+    private static FileReader createReader(URL from) throws FileDownloadException {
         return new FileReader(from);
     }
 
     private static FileWriter createWriter(File target) throws FileDownloadException {
-        assert target != null;
```
Comment (belkassaby) [inline src/main/java/org/monarchinitiative/biodownload/ftp/NIOftp.java:75]: Don't we want to make sure that target is not null?

**Review by belkassaby — APPROVED** [review]
I checked out this branch on my machine and all the tests are still running fine. (I tested the ignored tests too).

File: src/main/java/org/monarchinitiative/biodownload/ftp/NIOftp.java (hunk 1 above)
Comment (ielis) [reply src/main/java/org/monarchinitiative/biodownload/ftp/NIOftp.java:75]: It's never `null` as the method is private and it is null-checked before

Comment (ielis) [reply src/main/java/org/monarchinitiative/biodownload/ftp/NIOftp.java:75]: Moreover, assertions are disabled by default. So we should either do explicit null check with something like `Objects.requireNonNull`, or just leave it be. I think it should be OK to drop this assertion.

**Pull request merged**